lib/validators/liveValidator.ts

```diff
@@ -53,19 +40,48 @@ export interface Provider {
   [apiVersion: string]: ApiVersion
 }
 
+export interface RequestResponseObj {
+  readonly liveRequest: {
+    query: ParsedUrlQuery
+    readonly url: string
+    readonly method: string
+  }
+  readonly liveResponse: {
+    statusCode: string
+  }
+}
+
+export interface RequestValidationResult {
+  successfulRequest: Unknown
+  operationInfo?: Unknown
+  errors?: Unknown
+}
+
+export interface ResponseValidationResult {
+  successfulResponse: Unknown
+  operationInfo?: Unknown
+  errors?: Unknown
+}
+
+export interface ValidationResult {
+  readonly requestValidationResult: RequestValidationResult
+  readonly responseValidationResult: ResponseValidationResult
+  errors: Unknown[]
+}
+
 /**
  * @class
  * Live Validator for Azure swagger APIs.
  */
 export class LiveValidator {
-  public cache: {
+  public readonly cache: {
     [provider: string]: Provider
-  }
+  } = {}
```

Comment:
use the Map<T> ?

Reply:
BTW, I need to rename `Map` because `Map<K, T>` is already used by ES6. `MapObject` ?